### PR TITLE
[MM-27942] Request arch-specific plugin

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so when the server installs a plugin from the marketplace, it will request a bundle that is specific to the server's architecture. 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-27942

#### Related Pull Requests

https://github.com/mattermost/mattermost-marketplace/pull/116

### CODE REVIEW

File: model/marketplace_plugin.go [hunk 1]
```diff
@@ -95,6 +96,7 @@ func (filter *MarketplacePluginFilter) ApplyToURL(u *url.URL) {
 	q.Add("build_enterprise_ready", strconv.FormatBool(filter.BuildEnterpriseReady))
 	q.Add("enterprise_plugins", strconv.FormatBool(filter.EnterprisePlugins))
 	q.Add("local_only", strconv.FormatBool(filter.LocalOnly))
+	q.Add("arch", filter.Architecture)
```
Comment (lieut-data) [inline model/marketplace_plugin.go:99]: Would we consider sending up `OS` and `Arch` discretely instead?

Comment (ali-farooq0) [reply model/marketplace_plugin.go:99]: unless we call it something like `platform`?

**Review by lieut-data — APPROVED** [review]

**Review by hanzei — APPROVED** [review]

**Review by ali-farooq0 — APPROVED** [review]

**Comment by mickmister** [comment]
@prapti Given how small this change is, how do you feel about merging this? This is needed in order to speed up cloud deployments. There is no unit testing of the code around this change. I can make a ticket to write follow up tests.

Also, this change is backwards compatible with any marketplace version.

**Comment by prapti** [comment]
@mickmister Thank you for the context. Yeah, I can test this one post merge. Would it be possible to add a few QA test steps for testing later? That'd be great.

**Comment by mickmister** [comment]
@prapti Great, I've added some steps on the Jira ticket https://mattermost.atlassian.net/browse/MM-27942

**Pull request merged**